### PR TITLE
Support for @TemplateExtension matchName

### LINKED ISSUE #582: Support for @TemplateExtension matchName
Support for @TemplateExtension matchName

### PR DESCRIPTION
**Pull request by angelozerr**
Support for @TemplateExtension matchName

Fixes #582

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
It should start working:

image


I need tu support correctly matchName = ANY

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/project/datamodel/resolvers/FieldValueResolver.java [hunk 1]
```diff
@@ -67,6 +67,12 @@ public void setNamespace(String namespace) {
 		this.namespace = namespace;
 	}
 
+	@Override
+	public String getMatchName() {
+		// TODO Auto-generated method stub
```
Comment (fbricon) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/project/datamodel/resolvers/FieldValueResolver.java:72]: remove todo

**Comment by angelozerr** [comment]
> I need to support correctly matchName = ANY (config:*)

It should be supported correctly now:

ConfigAnyDemo

**Comment by fbricon** [comment]
property name showing an error 
Screenshot 2022-03-09 at 19 13 51

**Comment by angelozerr** [comment]
At first the syntax is wrong you need to use [] and this error is not linked to this pr.

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/project/datamodel/resolvers/FieldValueResolver.java (hunk 1 above)
Comment (angelozerr) [reply qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/project/datamodel/resolvers/FieldValueResolver.java]: Fixed

**Comment by angelozerr** [comment]
> This is an invalid syntax anyway. I think it should complain about the whole string though

Please create an issue for that

**Comment by fbricon** [comment]
Thanks @angelozerr

**Pull request merged**